backend/src/servers/servers_handler.rs

```diff
@@ -124,19 +139,23 @@ pub async fn update_server(
     
     let updated_server = server_service.update_server(&server_id, payload).await?;
 
+    let safe_aws = updated_server.aws_credentials.map(|creds| secure_server_response(creds));
+
+
     Ok(Json(ServerResponse {
```

Comment:
this code segment is duplicated, we can have a function to convert server to response and reuse it.